KIPs/kip-2.md

```diff
@@ -0,0 +1,259 @@
+               "cipherparams": {
+                   "iv": "92a89f3011e020f6a70bb34fddb006c5"
+               },
+               "cipher": "aes-128-ctr",
+               "kdf": "scrypt",
+               "kdfparams": {
+                   "dklen": 32,
+                   "salt": "aa38c864f4f6f538dc3bb4961c9d5d5772b163d2243aa8472cc4f83338e67cd3",
+                   "n": 4096,
+                    "r": 8,
+                    "p": 1
+                 },
+                "mac": "3355abc00f3fb1ae06821ece9c13359a0da1bd284cf29ffc26a534a40d210f28"
+            },
+            {
+               "ciphertext": '5e2f95f61d7af3bebf4ff9f5d5813690c80b0b5aaebd6e8b22d0f928ff06776a',
+               "cipherparams": {
+                   "iv": "92a89f3011e020f6a70bb34fddb006c5"
+               },
+               "cipher": 'aes-128-ctr',
+               "kdf": 'scrypt',
+               "kdfparams": {
+                   "dklen": 32,
+                   "salt": 'e7c4605ad8200e0d93cd67f9d82fb9971e1a2763b22362017c2927231c2a733a',
+                   "n": 4096,
+                   "r": 8,
+                   "p": 1,
+               },
+               "mac": 'fb86255428e24ba701201d5815f2f2114214cbd34fe4bc7a24b948a8ceac9f9b',
+            },
+        ]
+    }
+    ```
+3. Role-based keys
+    ```
+    {
+        "version": 4,
+        "id": "b6c6bd8f-e3ba-43c1-af4b-1b7dfe13e5f9",
+        "address": "0xa8677855999712d23692a4cad77bb9396a1174cd",
+        "keyring": [
+            [{
+                "ciphertext": "5cbd3cf235d02718f1cfac84157786d8e36fcf651e15f0c9a4303a2338697126",
+                "cipherparams": {
+                    "iv": "92a89f3011e020f6a70bb34fddb006c5"
+                },
+                "cipher": "aes-128-ctr",
+                "kdf": "scrypt",
+                "kdfparams": {
+                    "dklen": 32,
+                    "salt": "aa38c864f4f6f538dc3bb4961c9d5d5772b163d2243aa8472cc4f83338e67cd3",
+                    "n": 4096,
+                    "r": 8,
+                    "p": 1
+                },
+                "mac": "3355abc00f3fb1ae06821ece9c13359a0da1bd284cf29ffc26a534a40d210f28"
+            }],
+            [],
+            [
+                {
+                   "ciphertext": '5e2f95f61d7af3bebf4ff9f5d5813690c80b0b5aaebd6e8b22d0f928ff06776a',
+                   "cipherparams": {
+                       "iv": "92a89f3011e020f6a70bb34fddb006c5"
+                   },
+                   "cipher": 'aes-128-ctr',
+                   "kdf": 'scrypt',
+                   "kdfparams": {
+                       "dklen": 32,
+                       "salt": 'e7c4605ad8200e0d93cd67f9d82fb9971e1a2763b22362017c2927231c2a733a',
+                       "n": 4096,
+                       "r": 8,
+                       "p": 1,
+                   },
+                   "mac": 'fb86255428e24ba701201d5815f2f2114214cbd34fe4bc7a24b948a8ceac9f9b',
```

Comment:
```suggestion
                   "mac": "fb86255428e24ba701201d5815f2f2114214cbd34fe4bc7a24b948a8ceac9f9b",
```

Reply:
There are few single quoatation remain.
I tried to edit through suggested change, but I think it's a good idea to change it at once in your IDE, so i just leaved a comment.

Reply:
Got it. I updated them all. :)